match/lib/match/storage/git_storage.rb

```diff
@@ -135,9 +135,14 @@ def human_readable_description
       end
 
       def delete_files(files_to_delete: [], custom_message: nil)
-        # No specific list given, e.g. this happens on `fastlane match nuke`
-        # We just want to run `git add -A` to commit everything
-        git_push(commands: ["git add -A"], commit_message: custom_message)
+        if files_to_delete.count > 0
+          commands = files_to_delete.map { |filename|  "git rm #{filename}" }
+          git_push(commands: commands, commit_message: custom_message)
+        else
+          # No specific list given, e.g. this happens on `fastlane match nuke`
```

Comment:
Self-review:
idk if I should leave the old logic. All other storages expect a non-empty array of files to delete.

Reply:
not sure what you mean here.

Reply:
I took another look and decided that it would be better if git storage behaved like other storages and did not add unexpected additional changes to git if files_to_delete was empty.

Done in a474f7c5401de16000af7d2e8ed689d2ccd4e162.